yb-voyager/src/datafile/csvdatafile.go

```diff
@@ -30,29 +31,31 @@ func (df *CsvDataFile) SkipLines(numLines int64) error {
 }
 
 func (df *CsvDataFile) NextLine() (string, error) {
-	/*
-		// TODO: resolve the issue in counting bytes with csvreader
-		// here csv reader can be more useful in case filter table's columns
-		// currently using normal file reader will also work
+	var bytes bytes.Buffer
+	w := csv.NewWriter(&bytes)
+	w.Comma = []rune(df.Delimiter)[0]
+	var line string
+	var err error
+	for {
+		// read a record from csv file
 		fields, err := df.reader.Read()
 		if err != nil {
 			return "", err
 		}
 
-		line := strings.Join(fields, df.Delimiter)
-		df.bytesRead += int64(len(line + "\n")) // using the line in actual form to calculate bytes
-	*/
-
-	var line string
-	var err error
-	for {
-		line, err = df.reader.ReadString('\n')
+		// write the record to buffer and convert it to string following csv format
+		err = w.Write(fields)
```

Comment:
Perhaps in the next release, we can take latest golang version, which will allow us to use `InputOffset()` method (https://pkg.go.dev/encoding/csv@go1.19.5#Reader.InputOffset)